scheduler/src/cook/kubernetes/controller.clj

```diff
@@ -601,9 +613,10 @@
        (into {})))
 
 (defn scan-process
-  "Special verison of process run during scanning. It grabs the lock before processing the pod."
-  [{:keys [name] :as compute-cluster} pod-name]
-  (timers/time! (metrics/timer "scan-process" name)
-    (timers/time! (metrics/timer "process-lock" name)
-      (locking (calculate-lock pod-name)
-        (process compute-cluster pod-name)))))
+  "Given a pod-name, looks up the pod and delegates
+  to synthesize-state-and-process-pod-if-changed."
+  [{:keys [k8s-actual-state-map name] :as compute-cluster} pod-name]
+  (timers/time!
+    (metrics/timer "scan-process" name)
+    (let [{:keys [pod]} (get @k8s-actual-state-map pod-name)]
```

Comment:
Reading `k8s-actual-state-map` used to occur inside the lock.  Is there an issue reading from `k8s-actual-state-map`  before acquiring the lock?

Reply:
Good call. I moved the lock back out to callers of `synthesize-state-and-process-pod-if-changed` and put this `get` inside the lock.